tests/posix/c_lib_ext/src/fnmatch.c

```diff
@@ -30,8 +30,20 @@ ZTEST(posix_c_lib_ext, test_fnmatch)
 	zassert_equal(fnmatch("a*.c", "a/x.c", FNM_PATHNAME), FNM_NOMATCH);
 	zassert_ok(fnmatch("*/foo", "/foo", FNM_PATHNAME));
 	zassert_ok(fnmatch("-O[01]", "-O1", 0));
-	zassert_ok(fnmatch("[[?*\\]", "\\", 0));
-	zassert_ok(fnmatch("[]?*\\]", "]", 0));
+	if (!IS_ENABLED(CONFIG_COMMON_LIBC_FNMATCH)) {
+		/* \\ escapes ], no bracket expr, no match */
+		zassert_equal(fnmatch("[[?*\\]", "\\", 0), FNM_NOMATCH);
+		/* \\ escapes ], no bracket expr, no match */
+		zassert_equal(fnmatch("[]?*\\]", "]", 0), FNM_NOMATCH);
```

Comment:
This one is a little trickier.

Just mid-release, so give me a minute to revisit.